src/client/client.ts

```diff
@@ -143,6 +151,121 @@ class OAuth2 implements IOAuth2 {
+
+    const start = await this.client.startOAuth2Flow(
+      codeAcquisitionPattern,
+      timeout
+    );
+
+    // build our "proper" authorize URL using downstream implementation
+    const idpAuthorizeUrl = authorizeUrlFn({
+      state: start.state,
+      redirectUri: start.gatewayCallbackUrl,
+      codeChallenge: start.codeChallenge,
+    });
+
+    // formulate the authorize URL that the app user should use to that the grant flow
+    const authorizeUrl = new URL(start.gatewayAuthorizeUrl);
+    authorizeUrl.searchParams.set("return", idpAuthorizeUrl);
+    authorizeUrl.searchParams.set("callback", start.deskproCallbackUrl);
+
+    // curried poll function that returns the promise that resolves when polling is done... or rejects
+    // when polling times out
+    const poll = () => new Promise<OAuth2StartPollResult>((resolve, reject) => {
+      const poller = setInterval(() => {
+        this.client.pollOAuth2Flow(start.state).then((pollResult) => {
+          if (pollResult && pollResult.status !== "Pending") {
+            clearInterval(poller);
+            resolve({
+              status: pollResult.status,
+              error: pollResult.error,
+              authCodeProxyPlaceholder: pollResult.authCodeProxyPlaceholder,
+              codeVerifierProxyPlaceholder: pollResult.codeVerifierProxyPlaceholder,
+            } satisfies OAuth2StartPollResult)
+          }
+        });
+      }, pollInterval);
+
+      setTimeout(() => {
+        clearInterval(poller);
+        reject("Acquisition timeout");
+      }, timeout);
+    });
+
+    return {
+      poll,
+      authorizeUrl: authorizeUrl.toString(),
+    } satisfies OAuth2Start;
+  }
+
+  /**
+   *  {@inheritdoc}
+   */
+  async startStateless
+  (
+    authorizeUrlFn: OAuth2StartAuthorizeUrlFn,
+    codeAcquisitionPattern: RegExp,
+    options?: OAuth2StartOptions
+  ): Promise<OAuth2Start> {
+    const timeout = options?.timeout ?? (300 * 1000); // 5 minute timeout
+    const pollInterval = options?.pollInterval ?? 2000; // 2 second poll interval
+
+    const start = await this.client.startStatelessOAuth2Flow(
+      codeAcquisitionPattern,
+      timeout
+    );
+
+    // build our "proper" authorize URL using downstream implementation
+    const idpAuthorizeUrl = authorizeUrlFn({
+      state: start.state,
+      redirectUri: start.gatewayCallbackUrl,
+      codeChallenge: start.codeChallenge,
+    });
+
+    // formulate the authorize URL that the app user should use to that the grant flow
+    const authorizeUrl = new URL(start.gatewayAuthorizeUrl);
+    authorizeUrl.searchParams.set("return", idpAuthorizeUrl);
+    authorizeUrl.searchParams.set("callback", start.deskproCallbackUrl);
+
+    // curried poll function that returns the promise that resolves when polling is done... or rejects
+    // when polling times out
+    const poll = () => new Promise<OAuth2StartPollResult>((resolve, reject) => {
+      const poller = setInterval(() => {
```

Comment:
`setInterval` could cause a self DOS because it doesn't wait for the previous function a call to finish. So if you have slow/no internet or your computer goes to sleep, when it turns back on it "catches" backup and will send a bunch of requests. Instead, it's best to have a `setTimeout` inside the callee, so instead of calling every 5 seconds, it's 5 seconds _after_ the previous request finished.